_computing/resser_s3_from_R.md

```diff
@@ -0,0 +1,154 @@
+---
+title: Using S3 from R
+last_modified_at: 2018-07-09
+---
+
+# Using S3 from R
+
+You can use [Amazon Web Services' S3](https://aws.amazon.com/s3/) (Simple Storage Service) directly from `R`.
+
+The `R` package which facilitates this, `aws.s3`, is included in recent builds of `R` available on the `rhino` systems
+and the `gizmo` and `beagle` clusters.
+
+
+## Prerequisites
+
+### Get your AWS credentials
+
+Follow the ["Getting AWS Credentials"](https://teams.fhcrc.org/sites/citwiki/SciComp/Pages/Getting%20AWS%20Credentials.aspx) instructions on the SciComp wiki (must be on campus; sign in with HutchNet ID and password).
+
+This will create credentials (in the `.aws` directory of your home directory)
+that can be used by R (as well as Python and the AWS command line client)
+to authenticate your S3 requests.
+
+This page also explains what your PI bucket is called. 
+
+
+### Load an `R` module
+
+The first step is to load a recent R module:
+
+```
+ml R/3.5.0-foss-2016b-fh1
+```
+
+Then start R:
+
+```
+R
+```
+
+### Load the `aws.s3` package
+
+
+```R
+library(aws.s3)
+```
+
+**NOTE:** The example fragments from this point on assume you are in an `R` session with `aws.s3` loaded.
+
+### List all buckets
+
+```R
+bucketlist()
```

Comment:
Would it make sense to indicate handling returned values?  Like:

    blist <- bucketlist()

Reply:
Good idea, done.